docs/content/guides/dagster/local-to-production-tutorial.mdx

```diff
@@ -0,0 +1,420 @@
+                }
+            ),
+        },
+    }
+    deployment_name = os.environ.get("DAGSTER_DEPLOYMENT", "local")
+
+    return [
+        with_resources(
+            [items, comments, stories], resource_defs=resource_defs[deployment_name]
+        )
+    ]
+```
+
+Now, we can set the environment variable `DAGSTER_DEPLOYMENT=production` in our deployment and the correct resources will be applied to the assets.
+
+<Note>
+If you're using <a href="https://dagster.io/cloud">Dagster Cloud</a>, this environment variable will be set for you. Additionally, if you're using <a href="https://docs.dagster.cloud/guides/branch-deployments">Branch Deployments</a> <code>DAGSTER_DEPLOYMENT</code> will be automatically set to <code>VAR</code>.
+
+</Note>
+
+We still have some problems with this setup:
+
+1. Developers needs to remember to change `user` and `password` to their credentials and `schema` to their name when developing locally.
+2. Passwords are being stored in code.
+3. If we want to change the database where we store development or production data, we need to make a code change.
+
+We can easily solve all of these problems because the Snowflake IO manager accepts configuration from environment variables using the <PyObject object="StringSource" /> configuration type. This allows us to store configuration values as environment variables, and point Dagster to those environment variables in the run configuration:
+
+```python file=/guides/dagster/local_to_prod_tutorial/repository/repository_v5.py startafter=start endbefore=end
+# repository.py
+
+
+@repository
+def repo():
+    resource_defs = {
+        "hn_client": hn_api_client,
+        "io_manager": snowflake_io_manager.configured(
+            {
+                "account": {"env": "SNOWFLAKE_ACCOUNT"},
+                "user": {"env": "SNOWFLAKE_USER"},
+                "password": {"env": "SNOWFLAKE_PASSWORD"},
+                "database": {"env": "SNOWFLAKE_DATABASE"},
+                "schema": {"env": "SNOWFLAKE_SCHEMA"},
+            }
+        ),
+    }
+
+    return [*with_resources([items, comments, stories], resource_defs=resource_defs)]
+```
+
+<Note>
+
+Since all of the Snowflake IO manager configuration is being set through environment variables, we no longer need to select the resource configuration based on deployment! If we decide to add more configuration that varies between local and production, we can go back to the previous pattern of selecting config based on the deployment name.
+
+</Note>
+
+#### Staging
+
+Depending on your organization’s Dagster setup, there are a couple of options for a staging environment.
+
+If you are a Dagster Cloud user, we recommend using [Branch Deployments](https://docs.dagster.cloud/guides/branch-deployments) as your staging step. Check out our \[comprehensive guide to Branch Deployments]\(link to claire’s doc) to learn how to use Branch Deployments to verify your data pipelines before deploying them to production.
```

Comment:
It would be helpful to have a couple words that explain what a branch deployment is (i.e. a deployment that's automatically generated for each git branch you push).